[DOCS-1442] Self-Managed deployment guide revamp (#2113)
wandb/docs: platform/hosting/hosting-options/self-managed.mdx

Configuration follows a hierarchy: **Release Channel Values** (defaults from W&B), **User Input Values** (overrides via the System Console), and **Custom Resource Values** (your spec overrides both). For details, see [Deploy W&B with Kubernetes Operator](/platform/hosting/self-managed/operator).

- To deploy W&B Self-Managed in public cloud or on-premises infrastructure, see [Deploy W&B with Kubernetes Operator](/platform/hosting/self-managed/operator).
- To deploy W&B Self-Managed to a custom cloud platform that is not AWS, the requirements are similar to the requirements for deploying in [on-prem infrastructure](#deploy-wb-server-in-on-prem-infrastructure).
- To deploy W&B Self-Managed in air-gapped environments, see [Deploy on Air-Gapped Kubernetes](/platform/hosting/self-managed/on-premises-deployments/kubernetes-airgapped).

## Required infrastructure

> Regardless of where W&B Self-Managed is deployed, the deployment depends on the following infrastructure:

Reviewer: ```suggestion
All deployment options depend on the following infrastructure:
```

Author: Applied with a tiny correction
